docker/compose/faucet-validator-fullnode/docker-compose.yaml

```diff
@@ -0,0 +1,91 @@
+# This compose file supports running a faucet node that connects to the `validator-testnet`
+# compose's json-rpc endpoint. This should be started after that compose has been started.
+# The faucet can be accessed at 127.0.0.1:8000
+#
+# Place the following files into the same directory:
+# * mint.key (also known as the diem_root_key)
+# * waypoint.txt
+#
+# Additional notes:
+# * Images can be found at https://hub.docker.com/r/diem/faucet/tags, obtain the latest tag and
+# update below.
+version: "3.8"
+services:
+    validator:
+        # Note this image currently does not support this, will update to the appropriate minimum
+        # version shortly
+        image: libra/validator:devnet
+        networks:
+            shared:
+                ipv4_address: 172.16.1.3
+        volumes:
+            - type: bind
+              source: ./validator_data
+              target: /opt/diem/var
+        command: [ "/opt/diem/bin/diem-node", "--test", "--config", "/opt/diem/var" ]
+        ports:
+            - "8081:8080"
+
+    faucet:
+        image: libra/faucet:devnet
+        depends_on:
+            - validator
+        environment:
+            AC_HOST: "172.16.1.3"
+            AC_PORT: "8080"
+            CFG_CHAIN_ID: "TESTING"
+        networks:
+            shared:
+                ipv4_address: 172.16.1.2
+        volumes:
+            - type: bind
+              source: ./validator_data/mint.key
+              target: /opt/diem/etc/mint.key
+              read_only: true
+        command: >
+          /bin/bash -c "
+              sleep 3;
```

Comment:
we could also try the notion of restart on failure? or a while sleep loop?

Reply:
We probably should change faucet to lazy load mint key, so that start up won't fail when the key does not exist.

Reply:
the problem with faucet start really is the mint.key being blank. We could wait for the `mint.key` to be not blank, then start- that would work, sure

Reply:
I think it's fair to expect mint.key to exist when faucet is started, I think lazy loading of config is a good way to confuse users- if anything, I would like to validate perms on the key at startup as I was loading a key from before the rand update, and it was chugging along fine and at no point (even during minting) did it let me know there was an issue- mints would just fail in the cli and say "try again"

Reply:
Instead of blank files for these, could run a tiny alpine container or something first to mount the local bind directory  and create the files